tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs

```diff
@@ -245,11 +274,19 @@ internal static bool CheckRegistry(IRegistryService? registry = null, Version? t
                 bool ok = true;
 
                 // Check that the profiler is properly registered
-                if (tracerVersion == null || tracerVersion < new Version("2.14.0.0"))
+                if (tracerProgramVersion is null)
                 {
                     ok &= CheckClsid(registry, ClsidKey);
                     ok &= CheckClsid(registry, Clsid32Key);
                 }
+                else if (RuntimeInformation.OSArchitecture is Architecture.X64)
+                {
+                    ok &= CheckClsid(registry, ClsidKey);
```

Comment:
You may also want to check the 32-bit keys here, since the 64-bit installer will set both those registry locations

Reply:
Right I need to if the tracer is below `2.14` then it would set it, but after that we don't anymore right? Just wanted to confirm since I got a little lost but based on [this PR](https://github.com/DataDog/dd-trace-dotnet/pull/3549) I believe this is the case and we don't need to check for both starting on `2.14`.

Reply:
Hmmm we still set an inproc server for one of them here. I guess we'd have to see what the difference is between the 32-bit MSI and the 64-bit MSI: https://github.com/DataDog/dd-trace-dotnet/blob/5c41fe49801fa6b65e4b262ef7598c04dabe51c9/shared/src/msi-installer/shared/Files.wxs#L35

Reply:
So I ran a test when I saw that now we set the 32 key when running said installer and ever since 2.14 we only set the 64 bit ones if using the 64 bit installer, so I changed it to look like the below instead:
https://github.com/DataDog/dd-trace-dotnet/blob/maximo/cli-tool-improve-install-diagnosis/tracer/src/Datadog.Trace.Tools.Runner/Checks/ProcessBasicCheck.cs#L274-L291